### PR TITLE
feat(selector): add command option to toggle source selector

### PR DESCRIPTION
**Pull request by pysan3**
Closes: #1170, #895

Add command option `:Neotree selector=true / false` to toggle source selector on command.

Note: @cseickel 
This adds a new field `state.enable_source_selector` which I'm not sure if you are a fan of.

If you don't want to introduce a new field to `state`, I believe it would be pretty much impossible to implement this feature tho.

If you are OK with this PR, I'll add some documentation and change this from _draft_ to a PR.

### CODE REVIEW

**Comment by cseickel** [comment]
I think that's a nice clean solution. Go ahead and document it.

**Comment by pysan3** [comment]
> I think that's a nice clean solution. Go ahead and document it.

Thanks @cseickel. Updated the docs and is ready to be merged!

File: doc/neo-tree.txt [hunk 1]
```diff
@@ -128,6 +128,12 @@ want to change the current working directory. If you include this flag, it will
 automatically change the directory without prompting. This option implies
 "reveal", so you do not need to specify both.
 
+selector~
+This is a boolean flag. When you specifically set this to false (|selector=false|)
```
Comment (cseickel) [inline doc/neo-tree.txt:132]: The `|word|` syntax is for creating a link to a tag. Tags are created from words in this format:`*word*`. There is no `*selector=false*` tag so this is just a dead link. You probably want to use backticks here instead:

```suggestion
This is a boolean flag. When you specifically set this to false (`selector=false`)
```

File: doc/neo-tree.txt [hunk 2]
```diff
@@ -128,6 +128,12 @@ want to change the current working directory. If you include this flag, it will
...
+neo-tree will disable the |neo-tree-source-selector| for that neo-tree
+instance. Otherwise, the source selector will depend on what you specified in
+the configuration (|config.source_selector.{winbar,statusline}|).
```
Comment (cseickel) [inline doc/neo-tree.txt:135]: Same here, this should probably be backticks:
```suggestion
the configuration (`config.source_selector.{winbar,statusline}`).
```

File: doc/neo-tree.txt (hunk 1 above)
Comment (pysan3) [reply doc/neo-tree.txt]: Thanks. As you might have guessed, I'm very new to vim docs lol.
Fixed ;)

**Review by cseickel — APPROVED** [review]

**Pull request merged**